src/components/button/Button.tsx

```diff
@@ -8,24 +8,25 @@ import { setColor } from "../../utils/setcolor";
 import { Spinner } from "../spinner";
 import { SVGIcon, SVGIconProps } from "../svgicon";
 
-export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
-  children?: string | React.ReactNode;
-  severity?: "high" | "medium" | "low";
-  size?: "sm" | "md";
-  icon?: SVGIconProps["name"];
-  iconRight?: boolean;
-  initialState?: string;
-  color?: SingleColors;
-  form?: string;
-  select?: boolean;
-  className?: string;
-  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
-  as?: React.ElementType | "a" | "button";
-  linkTo?: string;
-  display?: "flex" | "inline-flex";
-  outline?: boolean;
-  secondary?: boolean;
-};
+export type ButtonProps<T extends React.ElementType = "button"> =
+  React.ButtonHTMLAttributes<HTMLButtonElement> & {
```

Comment:
Is this needed? Or left for backward-compatibility?

Reply:
It's not needed since the same is achieved by `React.ComponentProps<T>` for `T = "button"` - adjusted in 32a7b697551e575d698254f1ee5178d2b7e01272